metron-platform/metron-common/src/test/java/org/apache/metron/common/dsl/functions/StringFunctionsTest.java

```diff
@@ -32,124 +33,143 @@
-        thrown = false;
-
-        // NULL LENGTH
-        try{
-            run("FILL_RIGHT('123','X',bar)", variableMap);
-        }catch(ParseException pe) {
-            thrown = true;
-            Assert.assertTrue(pe.getMessage().contains("are both required"));
-        }
-        Assert.assertTrue(thrown);
-        thrown = false;
-
-        // NULL FILL
-        try{
-            run("FILL_RIGHT('123',foo, 7)", variableMap);
-        }catch(ParseException pe) {
-            thrown = true;
-            Assert.assertTrue(pe.getMessage().contains("are both required"));
-        }
-        Assert.assertTrue(thrown);
-        thrown = false;
-
-        // NON INTEGER LENGTH
-        try {
-            run("FILL_RIGHT('123','X', 'z' )", new HashedMap());
-        }catch(ParseException pe){
-            thrown = true;
-            Assert.assertTrue(pe.getMessage().contains("not a valid Integer"));
-        }
-        Assert.assertTrue(thrown);
-        thrown = false;
-
-        // EMPTY STRING PAD
-        try {
-            Object returnValue = run("FILL_RIGHT('123','', 10 )", new HashedMap());
-        }catch(ParseException pe) {
-            thrown = true;
-            Assert.assertTrue(pe.getMessage().contains("cannot be an empty"));
-        }
-        Assert.assertTrue(thrown);
-        thrown = false;
-
-        //MISSING LENGTH PARAMETER
-        try {
-            run("FILL_RIGHT('123',foo)", variableMap);
-        }catch(ParseException pe){
-            thrown = true;
-            Assert.assertTrue(pe.getMessage().contains("expects three"));
-        }
-        Assert.assertTrue(thrown);
+    Assert.assertTrue(thrown);
+    thrown = false;
+
+    // NULL FILL
+    try{
+      run("FILL_RIGHT('123',foo, 7)", variableMap);
+    }catch(ParseException pe) {
+      thrown = true;
+      Assert.assertTrue(pe.getMessage().contains("are both required"));
+    }
+    Assert.assertTrue(thrown);
+    thrown = false;
+
+    // NON INTEGER LENGTH
+    try {
+      run("FILL_RIGHT('123','X', 'z' )", new HashedMap());
+    }catch(ParseException pe){
+      thrown = true;
+      Assert.assertTrue(pe.getMessage().contains("not a valid Integer"));
+    }
+    Assert.assertTrue(thrown);
+    thrown = false;
+
+    // EMPTY STRING PAD
+    try {
+      Object returnValue = run("FILL_RIGHT('123','', 10 )", new HashedMap());
+    }catch(ParseException pe) {
+      thrown = true;
+      Assert.assertTrue(pe.getMessage().contains("cannot be an empty"));
+    }
+    Assert.assertTrue(thrown);
+    thrown = false;
+
+    //MISSING LENGTH PARAMETER
+    try {
+      run("FILL_RIGHT('123',foo)", variableMap);
+    }catch(ParseException pe){
+      thrown = true;
+      Assert.assertTrue(pe.getMessage().contains("expects three"));
     }
+    Assert.assertTrue(thrown);
+  }
+
+  @Test
+  public void shannonEntropyTest() throws Exception {
+    //test empty string
+    Assert.assertEquals(0.0, (Double)run("STRING_ENTROPY('')", new HashMap<>()), 1e-6);
```

Comment:
The specified equality delta is 1e-6, but is there a possibility that the expected return value for these cases is anything other than 0.0? I get it for imprecise double calculations, but I'm wondering if we would want to allow for that for these base cases.